src/js/video.js

```diff
@@ -89,6 +89,35 @@ var videojs = function(id, options, ready){
   return tag['player'] || new Player(tag, options, ready);
 };
 
+/**
+ * Expose private objects on the global videojs namespace using a Proxy if
+ * supported. Browsers that support Proxy will log a deprecation warning for
+ * directly accessing or modifying a deprecated object.
+ *
+ * @private
+ * @param {String} name The key name for the deprecated property.
+ * @param {Object} target The target object.
+ * @param {Object} messages String messages to display from a Proxy, mapping to get/set operations.
+ * @return {Object} A Proxy if supported or the `target` argument.
+ */
+const createDeprecationProxy = (name, target, messages) => {
```

Comment:
can we have this return either the proxy or just the target?
Then below in your usage we could do
```js
videojs.players = createDeprecationProxy(Players.players, {
  get: 'access to videojs.players is deprecated; use videojs.getPlayers instead',
  set: 'modification of videojs.players is deprecated'
});
```

Also, we should consider putting this function into its own file in `src/utils/` so that others could possibly use it as well.